### PR TITLE
Refactor Component Link Teaser

### PR DESCRIPTION
**Pull request by amponce**
**Description**

Refactor Component Link Teaser to allow for DRUPAL API data to get processed/transformed through data_service. This will allow the component to consume a basic object type.

**What was completed**

- [x]   Update component to use the new transformed data
- [x]   Create data_service to transform DRUPAL API and return new object
- [x]   Update Unit tests
- [x]   Update demo page
- [x]   Add to storybook

**Screenshots**
Screen Shot 2022-08-01 at 3 12 23 PM

Screen Shot 2022-08-01 at 3 15 58 PM

Tests:
```
 <LinkTeaser> component renders without field_spokes
    ✓ and click event sends correct params to recordEvent (48 ms)
    ✓ and without boldTitle (19 ms)
    ✓ and with boldTitle and title (11 ms)
    ✓ and with boldTitle and without title (9 ms)
  <LinkTeaser> component renders with field_spokes
    ✓ and click event sends correct params to recordEvent (23 ms)
    ✓ and without boldTitle (16 ms)
    ✓ and with boldTitle (14 ms)
    ✓ and without boldTitle and without title (7 ms)
  LinkTeaser with invalid data
    ✓ does not render <LinkTeaser> component when uri is not present (5 ms)

Test Suites: 1 passed, 1 total
Tests:       9 passed, 9 total
Snapshots:   0 total
Time:        1.665 s, estimated 2 s
```

### CODE REVIEW

**Comment by amponce** [comment]
Unfortunately the unit tests are not going to pass due to other errors.

File: src/components/link_teaser/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import Image from '@/components/image'
+import { get } from 'lodash'
+import { recordEvent } from '@/utils/recordEvent'
+import { ParagraphLinkTeaser } from '@/types/paragraph'
+
+export interface LinkTeaserProps extends ParagraphLinkTeaser {
```
Comment (tjheffner) [inline src/components/link_teaser/index.tsx:6]: i think we want to use this `ParagraphLinkTeaser` in the data service and not in the "pure" component. If the component props are still extending this it's going to expect all the drupal stuff along with it, isn't it? 

something like:

```js
export const transformLinkTeaserData = function (
  entity: ParagraphLinkTeaser,
  viewMode: string
): LinkTeaserProps { 
  // type check the entity is the right paragraph here
 
  return {
    ...mapping to the "pure" component props here
  }
}
```

**Review by tjheffner — CHANGES_REQUESTED** [review]

File: src/components/link_teaser/index.test.tsx [hunk 2]
```diff
@@ -225,25 +224,29 @@ describe('LinkTeaser with invalid data', () => {
...
       },
+      field_link_summary: 'Find out if you can get VA health care benefits.',
       drupal_internal__id: 123,
       drupal_internal__revision_id: 1,
       langcode: 'en',
       status: true,
     }
+
     const { container } = render(
       <Paragraph
```
Comment (penny-lischer) [inline src/components/link_teaser/index.test.tsx:244]: This needs to change to LinkTeaser

File: src/components/link_teaser/dataService.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { IMAGE_PATH_TEMP } from '@/lib/constants'
+import { ParagraphResourceType, ParagraphLinkTeaser } from '@/types/paragraph'
+import { EntityMetaInfo } from '@/lib/delegators/entityMetaProvider'
+import { LinkTeaser } from './index'
+
+export const transformAudienceTopicsData = function (
```
[the remark this answers is not in the document]
Comment (amponce) [reply src/components/link_teaser/dataService.ts:6]: Ahh, I copied the audience topics and missed that. Updating

File: src/components/link_teaser/index.tsx (hunk 1 above)
Comment (amponce) [reply src/components/link_teaser/index.tsx:6]: Ok great, I'll fix in the morning, I have a question for you about the componentParams type definitions

File: src/components/link_teaser/index.test.tsx (hunk 2 above)
Comment (amponce) [reply src/components/link_teaser/index.test.tsx]: Fixed

File: src/components/link_teaser/index.tsx (hunk 1 above)
Comment (amponce) [reply src/components/link_teaser/index.tsx]: Fixed!

File: src/components/link_teaser/dataService.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (amponce) [reply src/components/link_teaser/dataService.ts]: Fixed!

**Review by penny-lischer — APPROVED** [review]
LGTM

**Review by tjheffner — APPROVED** [review]

**Pull request merged**